### PR TITLE
Add latency formula

### PR DESCRIPTION
**Pull request by Oreoxmt**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v6.3 (TiDB 6.3 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs/pull/9763
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by Oreoxmt** [comment]
@you06 中文预览链接在[这里](https://preview-local.pingcap-docs-website-scaffold.pages.dev/zh/tidb/dev/latency-breakdown)

**Review by you06 — COMMENTED** [review]
And the `glossary.md` need to be updated.

**Review by you06 — APPROVED** [review]

File: latency-breakdown.md [hunk 1]
```diff
@@ -0,0 +1,874 @@
...
+```text
+tikv_grpc_msg_duration_seconds{type="kv_pessimistic_lock"} =
+    tikv_scheduler_latch_wait_duration_seconds{type="acquire_pessimistic_lock"} +
+    tikv_storage_engine_async_request_duration_seconds{type="snapshot"} +
+    (lock in-mem key count + lock on-disk key count) * lock read duration +
+    lock on-disk key count / (lock in-mem key count + lock on-disk key count) *
+    lock write duration
+```
+
+- 自 TiDB v6.0 起，TiKV 默认使用[内存悲观锁](/pessimistic-transaction.md#内存悲观锁)。内存悲观锁会跳过异步写入的过程。
```
Comment (ran-huang) [inline latency-breakdown.md:424]:
```suggestion
- 自 TiDB v6.0 起，TiKV 默认使用[内存悲观锁](/pessimistic-transaction.md#内存悲观锁)。内存悲观锁会跳过异步写入的过程。
- `tikv_storage_engine_async_request_duration_seconds{type="snapshot"}`是快照类型耗时，详情请参考 [TiKV 快照](#tikv-快照)小节.
```

File: latency-breakdown.md [hunk 2]
```diff
@@ -0,0 +1,874 @@
...
+                NonTerminal("Recv first"),
+            ),
+        ),
+    ),
+)
+```
+
+- 总体的发送请求耗时看作 `tidb_tikvclient_request_seconds`。
+- RPC 客户端为每个存储维护各自的连接池（称为 ConnArray），每个连接池都包含带有一个发送批量请求 channel 的 BatchConn
+- 批量请求会在存储是 TiKV 并且 batch 大小为正数时开启。这符合绝大多数的情况。
```
Comment (ran-huang) [inline latency-breakdown.md:610]:
```suggestion
- 绝大多数情况下，当存储是 TiKV 并且 batch 大小为正数时，批量请求开启。
```

File: latency-breakdown.md [hunk 3]
```diff
@@ -0,0 +1,874 @@
...
+- 批量请求 channel 的大小是 [`tikv-client.max-batch-size`](/tidb-configuration-file.md#max-batch-size)（默认值为 `128`）。请求入队的耗时看作 `tidb_tikvclient_batch_wait_duration`。
```
Comment (ran-huang) [inline latency-breakdown.md:611]:
```suggestion
- 批量请求 channel 的大小是 [`tikv-client.max-batch-size`](/tidb-configuration-file.md#max-batch-size) 的值（默认值为 `128`）。请求入队的耗时看作 `tidb_tikvclient_batch_wait_duration`。
```

File: latency-breakdown.md [hunk 4]
```diff
@@ -0,0 +1,874 @@
...
+- 一共有 `CmdBatchCop`、`CmdCopStream` 和 `CmdMPPConn` 三种流式请求。流式请求会引入一个额外的 `recv()` 调用来获取流中的第一个相应。
```
Comment (ran-huang) [inline latency-breakdown.md:612]:
```suggestion
- 一共有 `CmdBatchCop`、`CmdCopStream` 和 `CmdMPPConn` 三种流式请求。流式请求会引入一个额外的 `recv()` 调用来获取流中的第一个响应。
```

File: latency-breakdown.md [hunk 5]
```diff
@@ -0,0 +1,874 @@
...
+
+尽管有一些延迟没有被考虑，`tidb_tikvclient_request_seconds` 大致使用如下方式计算：
```
Comment (ran-huang) [inline latency-breakdown.md:614]:
```suggestion
`tidb_tikvclient_request_seconds` 大致使用如下方式计算（部分延迟不包含在内）：
```

File: latency-breakdown.md [hunk 6]
```diff
@@ -0,0 +1,874 @@
...
+        Sequence(
+            NonTerminal("Propose Wait"),
+            NonTerminal("Read index Read Wait"),
+        ),
+    ),
+    NonTerminal("Fetch A Snapshot From KV Engine"),
+)
+```
+
+一个 TiKV 快照的总体耗时可以使用 `tikv_storage_engine_async_request_duration_seconds{type="snapshot"}` 指标观察，它的计算方式如下：
```
Comment (ran-huang) [inline latency-breakdown.md:647]:
```suggestion
一个 TiKV 快照的总体耗时可以从 `tikv_storage_engine_async_request_duration_seconds{type="snapshot"}` 指标查看，它的计算方式如下：
```

File: latency-breakdown.md [hunk 7]
```diff
@@ -0,0 +1,874 @@
...
+
+### 慢读取查询
+
+如果 `SELECT` 语句占 Database Time 的很大一部分，你可以认为 TiDB 在读查询时速度很慢。
+
+慢查询的执行计划可以在 TiDB Dashboard 中的 [Top SQL 语句](/dashboard/dashboard-overview.md#top-sql-语句) 区域查看。要分析慢读取查询的耗时，你可以根据前面的描述分析[点查](#点查-point-get)、[批量点查](#批量点查-batch-point-get)和[表扫描和索引扫描](#表扫描和索引扫描-table-scan-和-index-scan)的耗时情况。
+
+### 慢写入查询
+
+在分析慢写入查询之前，你需要检查 `tikv_scheduler_latch_wait_duration_seconds_sum{type="acquire_pessimistic_lock"} by (instance)` 指标来确认冲突的原因：
```
Comment (ran-huang) [inline latency-breakdown.md:869]:
```suggestion
在分析慢写入查询之前，你需要查看 `tikv_scheduler_latch_wait_duration_seconds_sum{type="acquire_pessimistic_lock"} by (instance)` 指标来确认冲突的原因：
```

**Review by ran-huang — APPROVED** [review]

**Pull request merged**